src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -24,47 +29,53 @@ private sealed class RuleSetFile : IRuleSetFile
 
             private Exception _exception;
 
-            public RuleSetFile(string filePath, IVsFileChangeEx fileChangeService, VisualStudioRuleSetManager ruleSetManager)
+            public RuleSetFile(string filePath, VisualStudioRuleSetManager ruleSetManager)
             {
-                _filePath = filePath;
+                FilePath = filePath;
                 _ruleSetManager = ruleSetManager;
+            }
 
-                ImmutableArray<string> includes;
-
-                try
-                {
-                    includes = RuleSet.GetEffectiveIncludesFromFile(filePath);
-                }
-                catch (Exception e)
+            internal void InitializeFileTracking(IVsFileChangeEx fileChangeService)
```

Comment:
reason it being internal?

Reply:
Ah, right, that was from before I did the IReferenceCountedDisposable trick.